yarpl/include/yarpl/observable/Observables.h

```diff
@@ -47,6 +47,21 @@ class Observables {
     return Observable<T>::create(std::move(lambda));
   }
 
+  // this will generate an observable which can be subscribed to only once
+  template <typename T>
+  static Reference<Observable<T>> justOnce(T value) {
+    auto lambda = [value = std::move(value), used = false](Reference<Observer<T>> observer) mutable {
+      CHECK(!used) << "justOnce value was already used";
```

Comment:
I was going back and forth about this. On one hand I agree with you that crashing is not good. On the other hand this is violation of the API and the check is just checking that you are using it correctly. Debugging in the case of CHECK is much easier to debug.
I can also add DCHECK and exception instead of CHECK.
Anybody else want's to weigh in?